File: Algorithms/387-First-Unique-Character-in-a-String/First-Unique-Character-in-a-String.cpp
class Solution {
public:
    int firstUniqChar(string s)
    {
        unordered_map<char,int> hash_map;
        for(char c:s)
        {
            hash_map[c]++;
        }
        for (char c:s)
        {
            if (hash_map[c]==1)
            {
                return s.find(c);
            }
        }
        return -1;
    }
};

File: Algorithms/387-First-Unique-Character-in-a-String/First-Unique-Character-in-a-String.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        hash_map={}
        for i in s:
            hash_map[i]=hash_map.get(i,0) + 1
        for i,j in hash_map.items():
            if j==1:
                return s.find(i)    
        return -1    